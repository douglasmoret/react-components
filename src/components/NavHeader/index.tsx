import React, { useEffect } from 'react'
import classNames from 'classnames'
import style from './style.module.scss'
import NavIcon from 'components/NavIcon'
import capitalize from 'utils/capitalize'

interface NavHeaderProps {
  transparent?: boolean
  isFixed?: boolean
  className?: string
  openFrom?: 'top' | 'left'
  logoPath: string
  navOpened: boolean
  onNavClick: () => void
}

const NavHeader: React.FC<NavHeaderProps> = ({
  children,
  transparent = false,
  isFixed = false,
  className,
  openFrom = 'left',
  logoPath,
  navOpened,
  onNavClick
}) => {
  const containerClassnames = classNames(
    className,
    style.container,
    style[`openFrom${capitalize(openFrom)}`],
    {
      [style.opened]: navOpened,
      [style.isFixed]: isFixed,
      [style.transparent]: transparent,
    },
  )

  useEffect(() => {
    if (!window) {
      return
    }

    if (navOpened) {
      document.body.classList.add(style.scrollLocked)
    } else {
      document.body.classList.remove(style.scrollLocked)
    }

    return () => {
      document.body.classList.remove(style.scrollLocked)
    }
  }, [navOpened])

  return (
    <div className={containerClassnames}>
      <div className={style.header}>
        <div className={style.navContainer}>
          <div className={style.headerActions}>
            <NavIcon isCloseIcon={navOpened} className={style.navIcon} onClick={onNavClick} />

            <div className={style.headerLogo}>
              <figure>
                <img src={logoPath} />
              </figure>
            </div>
          </div>

          <nav className={style.nav}>
            <ul className={style.navList}>{children}</ul>
          </nav>
        </div>
      </div>
    </div>
  )
}

export default NavHeader
