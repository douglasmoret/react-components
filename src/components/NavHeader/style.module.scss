.scrollLocked {
  padding-right: calc(var(--spacing-md) - 1px);
  overflow-y: hidden;
}

.container {
  display: flex;
  align-items: center;
  justify-content: space-between;
  width: 100%;
  height: var(--size-header-height);
  position: static;
}

.header {
  width: 100%;
  box-shadow: var(--shadow-1);
  transition: background-color 200ms linear;
  background-color: var(--color-white);
}

.fixed .header {
  position: fixed;
  top: 0;
  left: 0;
  z-index: var(--depth-sidebar-nav);
}

.navContainer {
  height: var(--size-header-height);
  width: 100%;
  padding: var(--spacing-xs) var(--spacing-md);
  margin: 0 auto;
  display: flex;
  align-items: center;
  justify-content: space-between;
}

.headerActions {
  box-sizing: border-box;
  width: 100%;
  height: var(--size-header-height);
  display: flex;
  align-items: center;
}

.navIcon {
  margin-right: var(--spacing-md);
}

.opened .headerActions {
  position: fixed;
  top: 0;
  left: 0;
  padding: 0 var(--spacing-md);
  z-index: calc(var(--depth-sidebar-nav) + 1);
}

.headerLogo {
  max-width: 100px;
}

.headerLogo figure,
.headerLogo img,
.headerLogo svg {
  max-width: 100%;
  max-height: calc(var(--size-header-height) - (var(--spacing-xs)*2));
}

.opened .headerLogo svg {
  fill: var(--color-white);
}

.nav {
  position: fixed;
  top: 0;
  left: 0;
  width: 100%;
  height: 100vh;
  background-color: var(--color-white);
  display: flex;
  align-items: center;
  justify-content: center;
  transition: transform 225ms cubic-bezier(0, 0, 0.2, 1) 0ms;
  will-change: transform;
  z-index: var(--depth-sidebar-nav);
}

.openFromTop .nav {
  transform: translateY(-100%);
}

.openFromLeft .nav {
  transform: translateX(-100%);
}

.openFromTop.opened .nav {
  transform: translateY(0);
}

.openFromLeft.opened .nav {
  transform: translateX(0);
}

.navList {
  margin: 0;
  display: flex;
  flex-direction: column;
  align-items: flex-start;
  list-style: none;
  width: 100%;
  height: calc(100vh - var(--size-header-height));
  padding: var(--spacing-lg) var(--spacing-lg) 0;
}

@media only screen and (min-width: 1024px) {
  .navIcon {
    display: none;
  }

  .headerActions {
    width: unset;
    padding: unset;
    margin-right: var(--spacing-md);
  }

  .navContainer {
    width: var(--size-container-width);
    max-width: calc(100% - var(--spacing-lg));
    padding: var(--spacing-xs) var(--spacing-md);
  }

  .container .nav {
    position: unset;
    top: unset;
    left: unset;
    transform: unset;
    width: unset;
    height: unset;
    background-color: unset;
    flex: 1;
  }

  .navList {
    margin: 0;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: flex-start;
    padding: unset;
    height: unset;

  }
}
