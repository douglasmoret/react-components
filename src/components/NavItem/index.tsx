import React from 'react'
import classNames from 'classnames'
import Icon, { IconSlugsType } from '../Icon'
import Typography from '../Typography'
import style from './style.module.scss'

interface NavItemProps {
  className?: string
  icon: IconSlugsType
  label: string
  url: string
  onClick: () => void
  transparent?: boolean
  target?: '_blank' | '_self' | '_parent' | '_top'
  isActive: boolean
  element?: React.ElementType

}

const NavItem: React.FC<NavItemProps> = ({
  className,
  icon,
  label,
  url,
  onClick,
  children,
  transparent,
  target = '_self',
  isActive,
  element: Element,
  ...rest
}) => {
  const AnchorElement = Element || (url ? 'a' : 'span')
  const itemClassNames = classNames(style.navItem, className, {
    [style.transparent]: transparent,
  })

  if (children) {
    return <li className={itemClassNames}>{children}</li>
  }

  return (
    <li className={itemClassNames}>
      <AnchorElement
        onClick={onClick}
        target={target}
        className={style.navItemAnchor}
        href={url}
        {...rest}
      >
        <Icon className={style.navItemIcon} icon={icon} />
        <Typography
          element="span"
          variant="buttonMD"
          className={classNames(style.navItemLabel, {
            [style.navItemActive]: isActive,
          })}
        >
          {label}
        </Typography>
      </AnchorElement>
    </li>
  )
}

export default NavItem
