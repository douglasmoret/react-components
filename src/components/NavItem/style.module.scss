.navItem {
  margin: 0;
  display: flex;
}

.navItemAnchor {
  cursor: pointer;
  text-decoration: none;
  color: var(--color-primary-400);
  display: flex;
  align-items: center;
  height: var(--size-nav-item-height);
  text-transform: uppercase;
}

.navItemAnchor:hover {
  color: var(--color-primary-500);
}

.navItemIcon {
  margin-right: var(--spacing-sm);
}

.navItemIcon svg {
  fill: var(--color-primary-600);
}

.navItem .navItemLabel {
  align-self: center;
  color: var(--color-neutral-dark-300);
  text-decoration: none;
  font-family: var(--font-family);
  font-size: var(--font-size-xs);
  font-weight: var(--font-weight-bold);
  margin: unset;
}

.navItemActive {
  position: relative;
}

.navItemActive::after {
  content: '';
  width: 100%;
  height: 2px;
  display: block;
  background-color: var(--color-secondary-200);
  position: absolute;
  margin-top: calc(var(--spacing-sm) / 2);
}

@media only screen and (min-width: 1024px) {
  .navItemIcon {
    display: none;
  }

  .navItemAnchor {
    padding: var(--spacing-sm) var(--spacing-md);
  }

  .navItem .navItemLabel {
    padding-left: unset;
    transition: color 200ms linear;
    font-size: var(--font-size-xxs);
  }

  .navItemLabel:hover {
    color: var(--color-primary-500);
  }

  .navItemActive::after {
    background-color: var(--color-primary-300);
  }

}
