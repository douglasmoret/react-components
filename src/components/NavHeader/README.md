## Default

```js
import NavItem from '../NavItem'

const [navOpened, setNavOpened] = React.useState(false)
const switchNav = () => setNavOpened(!navOpened)

;<NavHeader
  transparent={false}
  navOpened={navOpened}
  onNavClick={switchNav}
  logoPath="http://d9hhrg4mnvzow.cloudfront.net/www.platosedu.com.br/28f9146a-asset-4_104q00r000000000000028.png"
  >
  <NavItem url="#navheader" label="ConheÃ§a a Platos" icon="mono_book" onClick={switchNav} />
  <NavItem url="#navheader" label="Perguntas frequentes" icon="mono_book" onClick={switchNav} />
  <NavItem url="#navheader" label="Baixe o app AVA" icon="mono_book" onClick={switchNav} />
</NavHeader>
```

### With transparent background

```js
import NavItem from '../NavItem'


const [navOpened, setNavOpened] = React.useState(false)
const switchNav = () => setNavOpened(!navOpened)

;<div style={{ backgroundColor: 'var(--color-primary-900)', padding: 16 }}>
  <NavHeader
    transparent
    navOpened={navOpened}
    onNavClick={switchNav}
    logoPath="http://d9hhrg4mnvzow.cloudfront.net/www.platosedu.com.br/28f9146a-asset-4_104q00r000000000000028.png"
  >
    <NavItem
      transparent
      url="#navheader"
      label="ConheÃ§a a Platos"
      icon="mono_book"
      onClick={switchNav}
    />

    <NavItem
      transparent
      url="#navheader"
      label="Perguntas frequentes"
      icon="mono_book"
      onClick={switchNav}
    />

    <NavItem
      transparent
      url="#navheader"
      label="Baixe o app AVA"
      icon="mono_book"
      onClick={switchNav}
    />
  </NavHeader>
</div>
```
