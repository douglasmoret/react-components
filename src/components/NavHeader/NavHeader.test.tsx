/* eslint-env jest */

import React from 'react'
import renderer from 'react-test-renderer'
import NavItem from '../NavItem'
import NavHeader from './index'

test('NavHeader: simple render test', () => {
  const component = renderer.create(
    <NavHeader
      navOpened={false}
      onNavClick={(): void => {
        console.log('clicked')
      }}
      isTransparent={false}
    >
      <NavItem url="#" label="conheÃ§a a Ahazou" icon="mono_book" />
      <NavItem url="#" label="perguntas frequentes" icon="mono_book" />
      <NavItem url="#" label="baixe o app Ahazou" icon="mono_book" />
    </NavHeader>
  )

  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
