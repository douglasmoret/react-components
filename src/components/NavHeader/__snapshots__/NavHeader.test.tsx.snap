// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavHeader: simple render test 1`] = `
<div
  className="container openFromLeft"
>
  <div
    className="header"
  >
    <div
      className="navContainer"
    >
      <div
        className="headerActions"
      >
        <div
          className="container navIcon"
        >
          <button
            className="icon"
            onClick={[Function]}
          >
            <span
              className="iconSpan"
            />
            <span
              className="iconSpan"
            />
            <span
              className="iconSpan"
            />
            <span
              className="iconSpan"
            />
          </button>
        </div>
        <div
          className="headerLogo"
        >
          <figure>
            <img
              src="http://d9hhrg4mnvzow.cloudfront.net/www.platosedu.com.br/28f9146a-asset-4_104q00r000000000000028.png"
            />
          </figure>
        </div>
      </div>
      <nav
        className="nav"
      >
        <ul
          className="navList"
        >
          <li
            className="navItem"
          >
            <a
              className="navItemAnchor"
              href="#"
              target="_self"
            >
              <span
                className="navItemIcon container md"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<svg width='16' height='20' viewBox='0 0 16 20'  ><path d='M14.4963 0H3.38853C1.97508 0 0.825195 1.15652 0.825195 2.57813V17.1875C0.825195 18.7383 2.07964 20 3.62156 20H14.4963C14.8395 20 15.1177 19.7202 15.1177 19.375C15.1177 19.0298 14.8395 18.75 14.4963 18.75H3.62156C2.76494 18.75 2.06803 18.0491 2.06803 17.1875C2.06803 16.3259 2.76494 15.625 3.62156 15.625H14.4963C14.8395 15.625 15.1177 15.3452 15.1177 15V0.625C15.1177 0.279844 14.8395 0 14.4963 0ZM2.06803 14.509V2.57813C2.06803 1.84461 2.65922 1.25 3.38853 1.25H3.42737C3.5346 1.25 3.62156 1.33746 3.62156 1.44531V14.1861C3.62156 14.2882 3.54338 14.3742 3.44205 14.3807C3.05037 14.4059 2.68039 14.5125 2.34867 14.6839C2.22039 14.7501 2.06803 14.654 2.06803 14.509Z' /></svg>",
                  }
                }
              />
              <span
                className="typography navItemLabel buttonMD"
              >
                conhe√ßa a Ahazou
              </span>
            </a>
          </li>
          <li
            className="navItem"
          >
            <a
              className="navItemAnchor"
              href="#"
              target="_self"
            >
              <span
                className="navItemIcon container md"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<svg width='16' height='20' viewBox='0 0 16 20'  ><path d='M14.4963 0H3.38853C1.97508 0 0.825195 1.15652 0.825195 2.57813V17.1875C0.825195 18.7383 2.07964 20 3.62156 20H14.4963C14.8395 20 15.1177 19.7202 15.1177 19.375C15.1177 19.0298 14.8395 18.75 14.4963 18.75H3.62156C2.76494 18.75 2.06803 18.0491 2.06803 17.1875C2.06803 16.3259 2.76494 15.625 3.62156 15.625H14.4963C14.8395 15.625 15.1177 15.3452 15.1177 15V0.625C15.1177 0.279844 14.8395 0 14.4963 0ZM2.06803 14.509V2.57813C2.06803 1.84461 2.65922 1.25 3.38853 1.25H3.42737C3.5346 1.25 3.62156 1.33746 3.62156 1.44531V14.1861C3.62156 14.2882 3.54338 14.3742 3.44205 14.3807C3.05037 14.4059 2.68039 14.5125 2.34867 14.6839C2.22039 14.7501 2.06803 14.654 2.06803 14.509Z' /></svg>",
                  }
                }
              />
              <span
                className="typography navItemLabel buttonMD"
              >
                perguntas frequentes
              </span>
            </a>
          </li>
          <li
            className="navItem"
          >
            <a
              className="navItemAnchor"
              href="#"
              target="_self"
            >
              <span
                className="navItemIcon container md"
                dangerouslySetInnerHTML={
                  Object {
                    "__html": "<svg width='16' height='20' viewBox='0 0 16 20'  ><path d='M14.4963 0H3.38853C1.97508 0 0.825195 1.15652 0.825195 2.57813V17.1875C0.825195 18.7383 2.07964 20 3.62156 20H14.4963C14.8395 20 15.1177 19.7202 15.1177 19.375C15.1177 19.0298 14.8395 18.75 14.4963 18.75H3.62156C2.76494 18.75 2.06803 18.0491 2.06803 17.1875C2.06803 16.3259 2.76494 15.625 3.62156 15.625H14.4963C14.8395 15.625 15.1177 15.3452 15.1177 15V0.625C15.1177 0.279844 14.8395 0 14.4963 0ZM2.06803 14.509V2.57813C2.06803 1.84461 2.65922 1.25 3.38853 1.25H3.42737C3.5346 1.25 3.62156 1.33746 3.62156 1.44531V14.1861C3.62156 14.2882 3.54338 14.3742 3.44205 14.3807C3.05037 14.4059 2.68039 14.5125 2.34867 14.6839C2.22039 14.7501 2.06803 14.654 2.06803 14.509Z' /></svg>",
                  }
                }
              />
              <span
                className="typography navItemLabel buttonMD"
              >
                baixe o app Ahazou
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</div>
`;
